sound/soc/sof/loader.c

```diff
@@ -20,21 +20,20 @@ static int get_ext_windows(struct snd_sof_dev *sdev,
 {
 	const struct sof_ipc_window *w =
 		container_of(ext_hdr, struct sof_ipc_window, ext_hdr);
-	size_t w_size = struct_size(w, window, w->num_windows);
 
 	if (w->num_windows == 0 || w->num_windows > SOF_IPC_MAX_ELEMS)
 		return -EINVAL;
 
 	if (sdev->info_window) {
-		if (memcmp(sdev->info_window, w, w_size)) {
+		if (memcmp(sdev->info_window, w, ext_hdr->hdr.size)) {
```

Comment:
@ktrzcinx why should we change this:? Is num_windows going to become obsolete?

Reply:
After change definition of `sof_ipc_struxt` this expression `struct_size(w, window, w->num_windows);` can't give correct size of received structure by contrast to `ext_hdr->hdr.size`

Reply:
ok

Reply:
@ktrzcinx is this correct? struct_size calculates "Calculate size of structure with trailing array". If num_windows is still correct, then the valid number of bytes in the window did not change. we don't need to copy the entire structure, do we?

Reply:
@pbossart https://github.com/thesofproject/linux/pull/2087#discussion_r422852863